Makefile

```
@@ -82,36 +82,65 @@ build-lcli-aarch64:
 build-lcli-riscv64:
 	cross build --bin lcli --target riscv64gc-unknown-linux-gnu --features "portable" --profile "$(CROSS_PROFILE)" --locked
 
-# extracts the current source date for reproducible builds
-SOURCE_DATE := $(shell git log -1 --pretty=%ct)
+# Environment variables for reproducible builds
+# Initialize RUSTFLAGS
+RUST_BUILD_FLAGS =
+# Remove build ID from the binary to ensure reproducibility across builds
+RUST_BUILD_FLAGS += -C link-arg=-Wl,--build-id=none
+# Remove metadata hash from symbol names to ensure reproducible builds
+RUST_BUILD_FLAGS += -C metadata=''
+
+# Set timestamp from last git commit for reproducible builds
+SOURCE_DATE ?= $(shell git log -1 --pretty=%ct)
+
+# Disable incremental compilation to avoid non-deterministic artifacts
+CARGO_INCREMENTAL_VAL = 0
+# Set C locale for consistent string handling and sorting
+LOCALE_VAL = C
+# Set UTC timezone for consistent time handling across builds
+TZ_VAL = UTC
+
+# Default features for lighthouse
+FEATURES ?= gnosis,slasher-lmdb,slasher-mdbx,slasher-redb,jemalloc
```

I think this breaks the `cli-check` due to this: https://github.com/chong-he/lighthouse/pull/8/commits/824a5dc735ec0babc9c7e4bdead18b3ef22aa461

I think we still support `mdbx` but it's going to be removed some time? We will wait for the devs to comment on this

---

what do you suggest to change it into? I am open for any suggestions that suits the purpose

---

We will wait for a review from @michaelsproul when he has the time

---

`jemalloc` is now the default

```suggestion
FEATURES ?= gnosis,slasher-lmdb,slasher-mdbx,slasher-redb
```

---

For simplicity, we could probably use the `CROSS_FEATURES` that's already defined in the `Makefile`? or is there something in `CROSS_FEATURES` that make the reproducible invalid?

---

Sure we can try using the cross features instead